### DISCUSSION TITLE
how can we find that if user cancelled their subscription in trial period in receipt?

### DISCUSSION THREAD

**Discussion by ghost** [imdhemy/appstore-iap#193] [Q&A]
how can we find that if user canceled their subscription in the trial period in the receipt?
here the user canceled their subscription in trial peroid but I can't find that in the receipt data
can anyone explain how to check in decoded receipt data ?

**Comment by imdhemy** [comment]
Hi @MihirPatel84 
Server notifications should handle this information. I think the following guide may be helpful. 
https://imdhemy.com/laravel-iap-docs/docs/server-notifications/app-store
